File: checkIns/src/main/java/com/popcorn/checkIns/outbox/OutboxEventRepository.java
package com.popcorn.checkIns.outbox;

import java.time.Instant;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface OutboxEventRepository extends JpaRepository<OutboxEvent, Long> {
    boolean existsByEventId(UUID eventId);

    int deleteByCreatedAtBefore(Instant cutoff);
}

File: payment/src/main/kotlin/com/popcorn/payment/outbox/OutboxEventRepository.kt
package com.popcorn.payment.outbox

import java.time.Instant
import java.util.UUID
import org.springframework.data.jpa.repository.JpaRepository

interface OutboxEventRepository : JpaRepository<OutboxEvent, Long> {
    fun existsByEventId(eventId: UUID): Boolean
    fun deleteByCreatedAtBefore(cutoff: Instant): Int
}
